File: Algorithms/diameter/diameter.c
typedef struct Node {
    struct Node* left;
    struct Node* right;
} Node;

int calcDiameter(Node* root) {
    // Your code here.
    return -1;
}

void test() {
    // Your tests here.
    Node* tree = (Node*)malloc(sizeof(Node));

    int result = calcDiameter(tree);
    printf("Result: %d\n", result);

    free(tree);
}

int main(int argc, char* argv[]) {
    test();
    return 0;
}

File: Algorithms/diameter/diameter.py
class Node:
  def __init__(self):
    self.left = None
    self.right = None

def calcDiameter(node):
  if node is None:
    return -1
  lh = getHeight(node.left)
  rh = getHeight(node.right)

  ld = calcDiameter(node.left)
  rd = calcDiameter(node.right)

  return max((lh+rh), ld, rd)

def getHeight(node):
  if node is None:
    return 0
  return max(getHeight(node.left), getHeight(node.right)) + 1

def main():
  root = Node()
  print(calcDiameter(root))
  root.left = Node()
  print(calcDiameter(root))
  root.right = Node()
  print(calcDiameter(root))
  root.left.right = Node()
  root.left.left = Node()
  print(calcDiameter(root))

if __name__ == '__main__':
  main()
